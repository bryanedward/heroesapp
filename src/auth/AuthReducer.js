import { types } from "../types/types";

const loginAuth = {
    types: types.login,
    payload: {
        name:'edward',
        password: '123'
    }
}

export const AuthReducer = ( state ={}, action) => {
    switch (action.type) {
        case types.login:
            return {
                ...action.payload,
                logged: true
            }            
            break;
        case types.logout:
            return {
                logged: false
            }
            break
    
        default:
            return state;
    }
}