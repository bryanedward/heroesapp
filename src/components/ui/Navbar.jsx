import React, { useContext } from 'react'
import { Link, NavLink, useNavigate } from 'react-router-dom'
import {AuthContext} from '../../auth/AuthContext'
import { types } from '../../types/types'
export const Navbar = () => {

  const {state} = useContext(AuthContext)
  const navigate = useNavigate()

  const {dispatch} = useContext(AuthContext)
  

  const action = {
    type: types.logout
  }

  const btnSalir = () => {
    dispatch(action)
    navigate('/login',{
      replace: true
    })
  }


  return (
    <nav className="navbar navbar-expand-sm navbar-dark bg-dark">

      <Link
        className="navbar-brand"
        to="/"
      >
        HeroApps
      </Link>

      <div className="navbar-collapse">
        <div className="navbar-nav">

          <NavLink
            className={({ isActive }) => 'nav-item nav-link' + (isActive ? 'active' : '')}
            to="/marvel">
            Marvel
          </NavLink>

          <NavLink
            className={({ isActive }) => 'nav-item nav-link' + (isActive ? 'active' : '')}
            to="/dc">
            DC
          </NavLink>
          <NavLink
            className={({ isActive }) => 'nav-item nav-link' + (isActive ? 'active' : '')}
            to="/heroe">
            Heroe
          </NavLink>
          <NavLink
            className={({ isActive }) => 'nav-item nav-link' + (isActive ? 'active' : '')}
            to="/search">
            Search
          </NavLink>
        </div>
      </div>

      <div className="navbar-collapse collapse w-100 order-3 dual-collapse2 justify-content-end">
        <ul className="navbar-nav ml-auto">
          <span className='nav-item nav-link text-info'>
            {state.name}
          </span>
          <button className="nav-item nav-link btn" onClick={btnSalir}>
            Logout
          </button>
        </ul>
      </div>
    </nav>
  )
}
