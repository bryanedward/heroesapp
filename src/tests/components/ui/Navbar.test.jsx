import {mount} from 'enzyme'
import { MemoryRouter, Route, Routes } from 'react-router-dom'
import { AuthContext } from '../../../auth/AuthContext'
import { Navbar } from '../../../components/ui/Navbar'
import { types } from '../../../types/types'

const mockNavigate = jest.fn()

jest.mock('react-router-dom',() => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate:() => mockNavigate

}))

describe('componente del navbar',() => {

    const contextValue = {
        dispatch: jest.fn( ),
        state: {
            name: 'brian',
            looged: false
        }
    }

    const wrapper = mount(<AuthContext.Provider value={contextValue}> 
        <MemoryRouter>
            <Routes>
                <Route path='/' element={<Navbar/>}/>
            </Routes>
        </MemoryRouter>
    </AuthContext.Provider>)

    test('validar el snapshot', () => { 
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('.text-info').text().trim()).toBe('brian')
    })

    test('debe de llamar de logout , llamar de navigate', () => { 
        wrapper.find('button').prop('onClick')()
        expect(contextValue.dispatch).toHaveBeenCalledWith({'type':types.logout})
        expect(mockNavigate).toHaveBeenCalledWith('/login',{replace:true})
     })
})