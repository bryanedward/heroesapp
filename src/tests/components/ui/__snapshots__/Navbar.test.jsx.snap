// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`componente del navbar validar el snapshot 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    href="/"
    onClick={[Function]}
  >
    HeroApps
  </a>
  <div
    className="navbar-collapse"
  >
    <div
      className="navbar-nav"
    >
      <a
        className="nav-item nav-link"
        href="/marvel"
        onClick={[Function]}
      >
        Marvel
      </a>
      <a
        className="nav-item nav-link"
        href="/dc"
        onClick={[Function]}
      >
        DC
      </a>
      <a
        className="nav-item nav-link"
        href="/heroe"
        onClick={[Function]}
      >
        Heroe
      </a>
      <a
        className="nav-item nav-link"
        href="/search"
        onClick={[Function]}
      >
        Search
      </a>
    </div>
  </div>
  <div
    className="navbar-collapse collapse w-100 order-3 dual-collapse2 justify-content-end"
  >
    <ul
      className="navbar-nav ml-auto"
    >
      <span
        className="nav-item nav-link text-info"
      >
        brian
      </span>
      <button
        className="nav-item nav-link btn"
        onClick={[Function]}
      >
        Logout
      </button>
    </ul>
  </div>
</nav>
`;
